### DISCUSSION TITLE
Delete Many

### DISCUSSION THREAD

**Discussion by kamalkech** [typegoose/typegoose#588] [Q&A] [answered]
```
"@typegoose/typegoose": "^7.4.5" :
"mongoose": "^5.10.9",

```
Service function  :

```ts
async deleteAll(): Promise<any> {
    try {
      return await this.blogModel
        .deleteMany({}, err => {
          return err;
        })
        .then(err => {
          if (err) {
            return false;
          }
          return true;
        });
    } catch (e) {
      const msg = `Exception deleting blogs. ${e}`;
      throw e;
    }
  }
```

Every time when run to delete all blogs i get this response object : 
`{ n: 0, ok: 1, deletedCount: 0 }`

but delete action work correctly and all data deleted,  and must to get `deletedCount: n`

Thank you to share if you have any idea, specially on the documentation anything found !

**Comment by hasezoey** [comment, marked as answer]
maybe try not to use callback and promise at the same time
